src/xtgeo/surface/_zmap_parser.py

```diff
@@ -0,0 +1,127 @@
+from functools import wraps
+import inspect
+from typing import Optional
+
+import numpy as np
+from pathlib import Path
+
+import dataclasses
+
+
+@dataclasses.dataclass
+class ZMAPSurface:
+    nrows: int
+    ncols: int
```

Comment:
In outher routines, the property names `ncol `and `nrow` are applied, not `ncols` and `nrows`